atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/asts/SweepableBucketRetriever.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.sweep.asts;
+
+import com.palantir.atlasdb.sweep.asts.SweepStateCoordinator.SweepableBucket;
+import java.util.List;
+
+public interface SweepableBucketRetriever {
+    /**
+     * Returns the sweepable buckets for a given shard. The list of sweepable buckets will be ordered by bucket identifier, then by shard.
+     * TODO: Should the ordering of the buckets be done here, or in another class?
```

Comment:
not actionable: I'm not sure the retriever has the context on how to prioritise this. So, this in theory could be `Set`. But, as discussed offline, given in initial work we likely want to use bucket identifier anyway, there's no reason to un-sort the buckets and then waste cycles sorting them again.